src/browser/BrowserService.cpp

```diff
@@ -314,7 +314,7 @@ QString BrowserService::getCurrentTotp(const QString& uuid)
     return {};
 }
 
-QJsonArray
+QPair<bool, QJsonArray>
 BrowserService::findEntries(const EntryParameters& entryParameters, const StringPairList& keyList, const bool httpAuth)
```

Comment:
since this function is so complex, would you rather pass in a `bool* accepted` and use that as your indicator? Set it to false at the top of the function (if non-null) and only set to true at the bottom. The problem with returning a QPair<bool, ...> is that I have no idea what the bool represents and how to handle it. Not a big deal for internal code, but it just isn't good practice and C++ offers no good solutions like Rust has.

I also think "httpAuth" should be embedded in EntryParameters

Reply:
This is exactly what I was also thinking about. Using `QPair` is pretty ugly here. Maybe the extra parameter could be a little bit more clear.

Reply:
especially if you move httpauth into EntryParameters